src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -413,8 +413,7 @@ int xdg_autostart_format_exec_start(
                         if (!escaped)
                                 return log_oom();
 
-                        free(exec_split[n]);
-                        exec_split[n++] = TAKE_PTR(escaped);
+                        free_and_replace(exec_split[n++], escaped);
```

Comment:
bug! free_and_replace() resolves the first argument twice, thus we'll have `n += 2` afterwards.

We could probably fix free_and_replace() by taking a ptr first, and operating on that, so to resolve things only once

Reply:
Pfff, I'll drop that patch from this PR. I thought it'd be a nice cleanup to do while looking that the quoting code, but it doesn't really belong here.